js/data/symbol_bucket.js

```diff
@@ -364,7 +366,7 @@ SymbolBucket.prototype.placeFeatures = function(collisionTile, buffers, collisio
         this.symbolInstances.sort(function(a, b) {
             var aRotated = sin * a.x + cos * a.y;
             var bRotated = sin * b.x + cos * b.y;
-            return aRotated - bRotated;
+            return (aRotated - bRotated) || (b.index - a.index);
```

Comment:
hmm, I don't think that would improve anything. It would just move the problem from numbers that have a difference of nearly zero to numbers that have a difference of nearly EPSILON.

Imagine two infinite precision values for `aRotated` and `bRotated` that when computed as floats could sometimes return true for `aRotated < bRotated` and could sometimes return false.  Your changes would fix the problem with these two values because `aRotated < bRotated + EPSILON` is always true. But now imagine a third infinite precision value `cRotated = bRotated - EPSILON`. The comparison `aRotated < cRotated` would not be consistent.

---

Doing something like `return (Math.round(aRotated) - Math.round(bRotated)) || (b.index - a.index);` would reduce the number of potentially inconsistent comparisons because only values around 0.5 could have problem. I could change it to that if you want, but I'm not sure that is even necessary.